rust/tools/authorization-logic/src/parsing/AuthLogic.g4

```diff
@@ -139,7 +145,9 @@ ATTRIBUTE: 'attribute';
 
 // Identifiers wrapped in quotes are constants whereas
 // identifiers without quotes are variables.
-ID : ('"')? [_a-zA-Z][_a-zA-Z0-9/.#:]* ('"')?;
```

Comment:
Oh, the old version currently allows unmatched `"` quotes in IDs. Is this desired, or should we use something like:
```
ID: RAW_ID | ( '"' RAW_ID '"' );
RAW_ID = [_a-zA-Z][_a-zA-Z0-9/.#:]*;
```

Reply:
You are right. That old version was a hack, and the new version fixes it. Fortunately, that change was still fixed in #593  which did get merged in :)

Reply:
(This PR existed only to save the very last commit which is the only way that this PR differs from #593 )